funcbench/main.go

```diff
@@ -99,35 +124,46 @@ The default is 2 hours (2h).`).Default("2h").Duration()
 				err error
 			)
 
-			if *gitHubPRNumber == 0 {
-				env, err = newLocalEnv(environment{
-					logger:        logger,
-					benchFunc:     *benchFunc,
-					compareTarget: *compareTarget,
-					home:          os.Getenv("HOME"),
-				})
+			// Setup Environment.
+			e := environment{
+				logger:        logger,
+				benchFunc:     cfg.benchFuncRegex,
+				compareTarget: cfg.compareTarget,
+			}
+			if cfg.ghPr == 0 {
+				// Local Mode.
+				env, err = newLocalEnv(e)
 				if err != nil {
-					return errors.Wrap(err, "new env")
+					return errors.Wrap(err, "environment creation error")
 				}
-				logger.Printf("funcbench start [Local Mode]: Benchmarking current version versus %q for benchmark funcs: %q\n", *compareTarget, *benchFunc)
 			} else {
-				if *owner == "" || *repo == "" {
-					return errors.New("funcbench in GitHub Mode requires --owner and --repo flags to be specified")
+				// Github Mode.
+				ghClient, err := newGitHubClient(ctx, cfg.owner, cfg.repo, cfg.ghPr, cfg.dryrun)
+				if err != nil {
+					return errors.Wrapf(err, "could not create github client")
 				}
-				env, err = newGitHubEnv(ctx, environment{
-					logger:        logger,
-					benchFunc:     *benchFunc,
-					compareTarget: *compareTarget,
-					home:          *workspace,
-				}, *owner, *repo, *gitHubPRNumber)
+
+				env, err = newGitHubEnv(ctx, e, ghClient, cfg.workspaceDir)
 				if err != nil {
-					return errors.Wrap(err, "new env")
+					if err := ghClient.postComment(fmt.Sprintf("%v. Could not setup environment, please check logs.", err)); err != nil {
+						return errors.Wrap(err, "could not post error")
+					}
+					return errors.Wrap(err, "environment creation error")
 				}
-				logger.Printf("funcbench start [GitHub Mode]: Benchmarking %q (PR-%d) versus %q for benchmark funcs: %q\n", fmt.Sprintf("%s/%s", *owner, *repo), *gitHubPRNumber, *compareTarget, *benchFunc)
 			}
 
 			// ( ◔_◔)ﾉ Start benchmarking!
-			return funcbench(ctx, env, newBenchmarker(logger, env, &commander{verbose: *verbose}, *benchTime, *benchTimeout, *resultCacheDir))
+			cmps, err := startBenchmark(ctx, env, newBenchmarker(logger, env, &commander{verbose: cfg.verbose}, cfg.benchTime, cfg.benchTimeout, cfg.resultsDir))
+			if err != nil {
+				if pErr := env.PostErr(fmt.Sprintf("%v. Benchmark failed, please check logs.", err)); pErr != nil {
+					return errors.Wrap(err, "could not log error")
```

Comment:
This is confusing. I think this has to be logged with `pErr`... wrapping it does not really make sense. It does not relate to error.